File: cpp/Problem242.cpp
/*
 * Problem: 242. Valid Anagram
 * URL: https://leetcode.com/problems/valid-anagram/
 * Difficulty: Easy
 * Tags: Array
 */

#include <string>

using namespace std;

class Solution {
public:
	bool isAnagram(string s, string t) {
		int letterFrequency[26] = {0};
		for (char &c : s) {
			letterFrequency[c - 'a']++;
		}
		for (char &c : t) {
			letterFrequency[c - 'a']--;
		}
		for (int i = 0; i < 26; i++) {
			if (letterFrequency[i]) {
				return false;
			}
		}
		return true;
	}
};

File: python/Problem242.py
# Problem: 242. Valid Anagram
# URL: https://leetcode.com/problems/valid-anagram/
# Difficulty: Easy
# Tags: Array

from typing import List

class Solution:
  def isAnagram(self, s: str, t: str) -> bool:
    letterFrequency: List[int] = [0] * 26
    for c in s:
      letterFrequency[ord(c) - ord('a')] += 1
    for c in t:
      letterFrequency[ord(c) - ord('a')] -= 1
    for frequency in letterFrequency:
      if frequency != 0:
        return False
    return True